index.js

```diff
@@ -48,11 +50,47 @@ function spawnClangFormat(args, done, stdio) {
               "Consider installing it with your native package manager instead.\n";
     throw new Error(message);
   }
-  var clangFormatProcess = spawn(nativeBinary, args, {stdio: stdio});
-  clangFormatProcess.on('close', function(exit) {
-    if (exit) done(exit);
-  });
-  return clangFormatProcess;
+
+  // extract glob, if present
+  var filesGlob = args.filter(function(arg){return arg.indexOf('--glob=') !== -1;})
+                    .map(function(arg){return arg.replace('--glob=', '');})
+                    .shift();
+
+  if (filesGlob) {
+    // remove glob from arg list
+    args = args.filter(function(arg){return arg.indexOf('--glob=') === -1;});
+
+    return glob(filesGlob, function(er, files) {
+      // launch a new process for each file, up to 200 at a time
+      async.parallelLimit(
+        files.map(function(file) {
+          return function(callback) {
+            var clangFormatProcess = spawn(nativeBinary,
```

Comment:
I don't think clang-format would have any problem iterating through a large number of files, but you do eventually hit the limit of a command-line argument length.
I think the simple answer is to partition the inputs into batches of ~100 and run clang-format on each batch, without parallelism. Then we don't have to deal with the tough questions like signal handling (ctrl-c kills the children correctly), error spam, etc